docs: add blog posts for npm-on-ipfs (#215)
ipfs/blog: content/post/74-putting-npm-on-ipfs-part-2.md

---

> date: 2019-03-04

Reviewer: ```suggestion
date: 2019-04-02
```

![Request sequence](/74-putting-npm-on-ipfs-part-2/ipfs-npm-sequence.png)

If this sounds familiar it may be because you've just read [part 1](/post/73-putting-npm-on-ipfs-part-1) - indeed it's the exact same code fulfilling the exact same function, just locally on your computer instead of on a server somewhere.

The http server is necessary because at the time of writing neither npm or Yarn support IPFS as a transport - hopefully one day this won't be necessary but as it stands no http traffic leaves your machine, unless you request modules that don't have CIDs in their metadata.

It also dials the ipfs-npm-registry-mirror directly to improve the speed of content resolution - this is partly because we know that the mirror will have the content we are after but also because there is no [DHT](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Distributed_hash_table.html) in js-IPFS (yet - it's coming!) so discovery would have to occur via the gateway nodes.

## 🎁 What's next?

So many things!

* We'd love (love!) to get IPFS into npm and Yarn as a first-class transport option

> * Once the DHT is enabled in js-ipfs v0.35.0 we'll not need to contact the npm mirror so a point of failure will be removed and startup will be faster

Reviewer: ```suggestion
* Once the DHT is enabled in js-ipfs v0.35.0 we'll not need to contact the npm mirror so that single point of failure will be removed and startup will be faster
```

> * Once the DHT is enabled in js-ipfs v0.35.0 we'll not need to contact the npm mirror so a point of failure will be removed and startup will be faster

Reviewer 2: ```suggestion
* Once the DHT is enabled in js-ipfs v0.36.0 we'll not need to contact the npm mirror so a point of failure will be removed and startup will be faster
```

* How about a service you can run on on your local network to further speed up package resolution with no need to get onto the Internet?

> Publishing? Identity? The posibilities are endless - if you'd like to help out please visit [ipfs-shipyard/npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs).

Reviewer: ```suggestion
Publishing? Identity? The posibilities are endless - if you'd like to help out please visit [ipfs-shipyard/npm-on-ipfs](https://github.com/ipfs-shipyard/npm-on-ipfs), or check out our [📦 IPFS Package Managers Working Group](https://github.com/ipfs/package-managers).
```
